03week/ticTacToe.js

```diff
@@ -25,22 +45,112 @@ function printBoard() {
 
 function horizontalWin() {
   // Your code here
+  let foundWin = false
+  board.forEach(row =>{
+    if(foundWin){
+      //if something is true, stop next loop
+      return
+    }
+    let counter = 0
+    row.forEach(space =>{
+      if(space === playerTurn){
+        counter = counter + 1
+      }
+    })
+    if(counter === 3){
+      //console.log('counter is 3')
+      foundWin = true
+    }
+  })
+  return foundWin
 }
 
 function verticalWin() {
   // Your code here
+  const vertLineWins = [
+    [board[0][0], board[1][0], board[2][0]],
+    [board[0][1], board[1][1], board[2][1]],
+    [board[0][2], board[1][2], board[2][2]]
+  ]
+  let foundWin = false
+  vertLineWins.forEach(possibleWin => {
+    if (foundWin) {
+      //if something is true, stop next loop
+      return
+    }
+    let counter = 0
+    possibleWin.forEach(space => {
+      if (space === playerTurn) {
+        counter = counter + 1
+      }
+    })
+    if (counter === 3) {
+      //console.log('counter is 3')
+      foundWin = true
+    }
+  })
+  return foundWin
 }
 
 function diagonalWin() {
   // Your code here
+  const slashWins = [
+    [board[0][0],board[1][1],board[2][2]],
+    [board[0][2],board[1][1],board[2][0]]
+  ]
+  let foundWin = false
+  slashWins.forEach(possibleWin => {
+    if (foundWin) {
+      return
+    }
+    let counter = 0
+    possibleWin.forEach(space => {
+      if (space === playerTurn) {
+        counter = counter + 1
+      }
+    })
+    if (counter === 3) {
+      foundWin = true
+    }
+  })
+  return foundWin
 }
 
 function checkForWin() {
   // Your code here
+  if(horizontalWin() || diagonalWin() || verticalWin()){
+    console.log('Congrats player ' + playerTurn + ' you won!')
+    return true
+  }
+
+}
+
+function displayMove(row, column){
+  board[row][column] = playerTurn;
+}
+
+function isValidMove(row, column) {
+  if (board[row][column] === ' '){
+    return true;
+  }else {
+    return false;
+  }
 }
 
 function ticTacToe(row, column) {
   // Your code here
+  if(isValidMove(row, column)){
```

Comment:
isValidMove should also make sure that the input that is entered is also valid. For example i should not be able to input other letters, numbers, and negative numbers as a input.